src/mavsdk/plugins/info/info_impl.cpp

```diff
@@ -46,17 +48,16 @@ void InfoImpl::deinit()
 
 void InfoImpl::enable()
 {
-    // We can't rely on System to request the autopilot_version,
-    // so we do it here, anyway.
-    _system_impl->send_autopilot_version_request();
```

Comment:
Shouldn't this one stay for PX4?

Reply:
So the deadlock I mentioned didn't seem Ardupilot specific. Also, this message is called immediately before this is enabled [here](https://github.com/mavlink/MAVSDK/blob/7b30f7383934823ae564ed23788d0f2fc1dfa47c/src/mavsdk/core/system_impl.cpp#L582) anyways. This happens whenever info plugin is created too quickly after the system.

Reply:
Okay thank you, you just reminded exactly what the problem is. This is the only plugin you can't initialize in a callback.